File: 050-059/ABC055_B.c
// ABC055 B - Training Camp
#include <stdio.h>
#include <math.h>

int main(void) {
    int N;
    unsigned long long M;
    unsigned long long result = 1;

    scanf("%d", &N);
    M = (unsigned long long)pow(10.0, 9.0) + 7;
    unsigned long long i;
    for (i = 1; i<= N; i++) {
        result = ((result % M) * (i % M)) % M;
    }
    printf("%lld\n", result);

    return 0;
}

File: 050-059/ABC055_B.js
// ABC055 B - Training Camp
'use strict';

function Main(input) {
    const tmp = input.trim().split('\n');
    const N = Number(tmp[0]);
    // console.log(N);

    const M = Math.pow(10, 9) + 7;

    let result = 1;
    for (let i = 1; i <= N; i++) {
        result = ((result % M) * (i % M)) % M;
    }
    console.log(result);
}

Main(require('fs').readFileSync('/dev/stdin', 'utf8'));

// Main(`3`);
// Main(`10`);
// Main(`100000`);
